### DISCUSSION TITLE
Why doesn't validateAndCreate use the validated data?

### DISCUSSION THREAD

**Discussion by piotrjoniec** [spatie/laravel-data#638] [General]
Currently, the `validateAndCreate` method validates the data, then creates the data object from the unfiltered payload:

```php
    public static function validateAndCreate(Arrayable|array $payload): static
    {
        static::validate($payload);

        return static::from($payload);
    }
```

With the current approach, the "exclude" validation rules don't work. Why not just do this:

```php
    public static function validateAndCreate(Arrayable|array $payload): static
    {
        return static::from(static::validate($payload));
    }
```

Now the exclude rules are working. Am I missing something?

**Comment by rubenvanassche** [comment]
Yeah this is something I want to do, and an open issue on my list. Problem is that this causes a lot of issues with mapping and such, something we're going to take a look at in v5.
